### ISSUE TITLE
Load sidebar only on demand

### ISSUE THREAD

**Issue by niutech** [powerpak/hn-sidebar#4]
Privacy is key, so sending all the visited URLs to any third-party is inacceptable. Why not showing the sidebar only after clicking the button, so that only the page I want will be searched for comments?

**Comment by powerpak** [comment]
Good idea; i should note that the current design is that the button _only_ shows if there is an HN thread about it.  That's a nice datapoint, because it often shows for pages that I wouldn't expect to have found a thread for, and then you can scan the comments.  And when it doesn't, I know there probably isn't such a thread.

If it shows up on all pages regardless of whether there is a thread, that might be distracting, because you'll wind up clicking on it all the time just to check whether there is a thread.

**Comment by huyz** [comment]
+1 I'd rather preserve my privacy and search for a HN thread only when I'm interested in comments, in the same way I sometimes make the conscious decision to jump to the bottom of an article looking for in-page comments.

**Comment by huyz** [comment]
I wouldn't always show a floating HTML button on every page. I would have an extension icon in the Chrome toolbar which I would click as needed.

**Comment by huyz** [comment]
I modified the extension to enable on-demand search for HN:
https://github.com/huyz/hn-sidebar

I didn't create a pull request because it should be a user option to have one behavior or the other and I didn't implement that toggle.
